### ISSUE TITLE
deleting G187 ?

### ISSUE THREAD

**Issue by goetsu** [w3c/wcag#292]
https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20161007/G187

is allowing author to rely on browser to stop animated content in a certain technology and specially make reference to gif and apng.

Last test I've  done show that none of the major browsers currently stop gif anymore using esc key, Chrome and firefox allow it with a browser extension, no solution on edge or safari.

So I'm wondering if we can still consider this as a viable technique to conform 2.2.2 at least for gif and apng.

**Comment by goetsu** [comment]
as @detlevhfischer also mention on twitter, esc key will not work on mobile / tablet (but I think some browser show a play / pause button on animated gif)

**Comment by DavidMacDonald** [comment]
> Last test I've done show that none of the major browsers currently stop gif anymore using esc key, Chrome and firefox allow it with a browser extension, no solution on edge or safari.

Agree... just had that issue with a client yesterday... have to install pause button.

**Comment by goetsu** [comment]
@DavidMacDonald same problem here but will be hard to impose a pause button (specially for a animated gif) when I still have this technique listed as a possible solution

**Comment by awkawk** [comment]
The technique has a test procedure, and if the browsers fail the test procedure
then people can’t rely on that technique. Note also that the technique does
not require hitting Esc to work - a browser preference or extension would be ok
also.

It may be that support is lacking in Edge and Safari and in mobile browsers so that the technique is effectively not able to be used, but we could easily add a note to raise the caution. I'm worried that this is a distraction from WCAG 2.1 work, so this isn't a very high priority.

**Comment by goetsu** [comment]
@awkawk  if what / how many browsers fail test procedure ? as wcag don't impose any browsers stack if it work on at least 1 browser they still can pretend to be conforming

**Comment by awkawk** [comment]
@goetsu and they can do that even if the technique doesn't exist.

**Comment by goetsu** [comment]
@awkawk no because it will fail 2.2.2 in that case or i'm missing something

**Comment by detlevhfischer** [comment]
Agree with @awkawk that it is not high priority - but a decision to retire this Technique seems easy if support is as bad as indicated.

**Comment by awkawk** [comment]
A official technique doesn't need to be present to pass an SC.

**Comment by goetsu** [comment]
@awkawk yes but a least if no technique say that using esc key is a possible solution they can't use it as a counter argument when someone ask for a pause button or other solution that can be used by real users

**Comment by accessiblewebuk** [comment]
Following the discussion on Twitter I would like to add that a visible control would also be needed for voice recognition / activation (e.g. Dragon) users.

**Comment by Veyfeyken** [comment]
+1 for deletion

**Comment by cyns** [comment]
I came here to suggest clarifying that browser suppression of animation is a mechanism, which was the original intent of that term. Using an extension that stops animation should be a technique. I see that this issue wants to go in the opposite direction. As the author of the original text, I disagree.

**Comment by cyns** [comment]
In general, it seems that Chrome has all of its accessibility features as extensions. That should not prevent authors from relying on those features. Allowing users to globally set things in their operating system and browser is always better than requiring every website to do it. Each website will do it in a different way, which can add to confusion.

**Comment by cstrobbe** [comment]
This technique dates to the time when people still remembered browsers that stopped animated GIFs when you pressed Esc. I'll quote*Constructing Accessible Web Sites* by Jim Thatcher et al (glasshaus, 2002; page 107):

> The two major browsers (Microsoft Internet Explorer and Netscape Navigator) permit uses to stop animation in GIF's by pressing the escape key. Opera more effectively allows users to turn off animation by using the menu options `File | Preferences`, there in the *Multimedia* section uncheck the checkbox labeled "Enable Animation (GIF images)".

Internet Explorer 5 is no longer in use and Netscape Navigator was succeeded by Firefox. Cheah Chu Yeow's book *Firefox Secrets* (SitePoint, 2005) makes no mention of the escape key to stop animated GIFs. Page 241 points out that Chris Pederick's Web Developer Toolbar had a feature named "Disable Image Animations" (in its "Disable" menu), but this feature has long disappeared.

I am currently not aware of user agents that allow you to stop an animation by simply clicking a specific key or a keyboard combination. It usually requires installing browser extensions, client-side scripting or changing obscure settings in browsers. For Chrome, there is the extension [GIF Blocker](https://chrome.google.com/webstore/detail/gif-blocker/ahkidgegbmbnggcnmejhobepkaphkfhl). In Firefox, you need to go to `about:config`, search ` image.animation_mode` and set its value to "once" or "none". (See [How does one turn off animated GIFs](https://support.mozilla.org/en-US/questions/1350240).)

@cyns wrote,
> Using an extension that stops animation should be a technique.

Technologically adept users like us know about browser extensions and can install them. This is not true for the general population. And you aren't necessarily install browser extensions on computers in libraries etc.

@awkawk wrote,
> It may be that support is lacking in Edge and Safari and in mobile browsers so that the technique is effectively not able to be used (...)

I would venture to say that support won't be added to modern browsers because too many ads on web pages rely on animated GIFs. (In fact, that may even be the reason why some browsers stopped supporting the ESC key to stop animations.)

**Comment by GreggVan** [comment]
If this is the case (browsers no longer have build in features to stop looping animated gifs )— then I think authors need to build "stop animation’ into their page for looping or very long animations.   So I agree we should not keep this technique around since it is no longer valid - that is, the technique no longer works.

And I don’t think it is reasonable for us to assume users know how to install extensions — or that they will be installed on all of the computers in all of the locations that a person needs to view the internet. 

Best   So +1 for removing technique


Gregg Vanderheiden
***@***.***
